### PR TITLE
feat(TreeView): publish single select tree

### LINKED ISSUE #11874: Bring single select tree view to `stable`
steps taken from draft checklist in https://github.com/carbon-design-system/carbon/issues/11688

* [x]  All files have a copyright banner
* [x]  All components exported in `src/index.js` and should not be `unstable_` prefixed
* [x]  Component has a label in the GitHub repository
* [ ]  Component should be documented on the website
  * [x]  Component should have a [usage](https://github.com/carbon-design-system/carbon-website/issues/2210), [style](https://github.com/carbon-design-system/carbon-website/issues/2211), and [code](https://github.com/carbon-design-system/carbon-website/issues/2216) tab
  * [ ]  Component may have a [component demo](https://github.com/carbon-design-system/carbon-website/issues/3112)
* [ ]  For each component exported:
  * [x]  Component is written as a function declaration or uses `forwardRef`
  * [x]  Component has `propTypes` defined
    * [x]  Each prop type has a comment (used in storybook)
    * [x]  Prop types are as specific as needed, prefer `PropTypes.shape` over `PropTypes.object` if possible
  * [ ]  Default props are listed as default args in the function definition (not in defaultProps)
    * [ ]  Note: default props should be stable, in other words props like `onClick => () ={}` can cause re-renders since the function identity is not stable
  * [ ]  Component has a story in `<ComponentName>.stories.js`
    * [ ]  Component has an mdx document that follows our outline
    * [ ]  mdx document coverages at least common use-cases and provides a prop table
    * [x]  Stories cover at least common use-cases
    * [x]  Stories may include a `Playground` story for controls
    * [x]  Stories should mirror intended usage of the component
  * [ ]  Component has unit/integration tests written in RTL for testing the component API
  * [x]  Component is tested via VRT for at least the initial render state
  * [x]  Component is tested via AVT for at least the initial render state

fixes and enhancements potentially independent of single select tree release

* [ ] #11858
* [ ] #11859

### PR DESCRIPTION
**Pull request by emyarod**
Closes #11874
ref #11858

This PR adds additional tests and marks the single select treeview component as stable. The multiselect prop will remain experimental

documentation updates will depend on https://github.com/carbon-design-system/carbon-website/pull/3062

#### Changelog

**New**

- treeview e2e testfile

**Changed**

- add experimental label to `multiselect` prop

**Removed**

- remove `unstable_` prefix

### CODE REVIEW

**Comment by tay1orjones** [comment]
Is the vertical alignment of the arrow correct? Based on the docs PR it looks like it should be vertically centered within the node, but here it's a bit above-center

image

**Comment by tay1orjones** [comment]
I just pushed an update with the following:

* Remove `-story.js` file in favor of `-stories.js` file
* Add Playground story and associated controls
* Remove args/controls from non-playground stories
* Remove v11 feature flags in treeview source

File: e2e/components/TreeView/TreeView-test.e2e.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          theme,
+        });
+      });
+    });
+  });
+
+  test('accessibility-checker @avt', async ({ page }) => {
+    await visitStory(page, {
+      component: 'TreeView',
+      id: 'components-treeview--treeview-story',
```
Comment (tay1orjones) [inline e2e/components/TreeView/TreeView-test.e2e.js]: I think this is why the percy status check is failing

```suggestion
      id: 'components-treeview--default',
```

File: e2e/components/TreeView/TreeView-test.e2e.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+const { themes } = require('../../test-utils/env');
+const { snapshotStory, visitStory } = require('../../test-utils/storybook');
+
+test.describe('TreeView', () => {
+  themes.forEach((theme) => {
+    test.describe(theme, () => {
+      test('treeview @vrt', async ({ page }) => {
+        await snapshotStory(page, {
+          component: 'TreeView',
+          id: 'components-treeview--treeview-story',
```
Comment (tay1orjones) [inline e2e/components/TreeView/TreeView-test.e2e.js]:
```suggestion
          id: 'components-treeview--default',
```

**Review by tay1orjones — APPROVED** [review]
Looks great! 🎉

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Review by laurenmrice — APPROVED** [review]

**Pull request merged**